app/code/Magento/Sales/Model/ResourceModel/EntityAbstract.php

```diff
@@ -176,7 +176,11 @@ protected function _afterSave(\Magento\Framework\Model\AbstractModel $object)
      */
     protected function updateObject(\Magento\Framework\Model\AbstractModel $object)
     {
-        $condition = $this->getConnection()->quoteInto($this->getIdFieldName() . '=?', $object->getId());
+        $idFieldValue = $this->getConnection()->prepareColumnValue(
+            (array)$this->_getColumnDefinition($this->getIdFieldName(), $this->getMainTable()),
+            $object->getId()
+        );
+        $condition = $this->getConnection()->quoteInto($this->getIdFieldName() . '=?', $idFieldValue);
```

Comment:
quoteInto() will return quoted value in any case, value after prepareColumnValue() will be quoted anyway, that makes this value preparation little bit useless.